src/transaction/settings.js

```diff
@@ -87,15 +88,18 @@ function createSettingsTransaction(account: string, settings: Settings
     TransactionType: 'AccountSet',
     Account: account
   };
+
+  if (settings.memos !== undefined) {
+    txJSON.Memos = _.map(settings.memos, utils.convertMemo);
+    delete settings.memos;
```

Comment:
Instead of mutating, how about just passing `_.omit(settings, 'memos')` to `setTransactionFlags`?